### ISSUE TITLE
Support for multiple/different API providers in a single Chopper client

### ISSUE THREAD

**Issue by chimon2000** [lejard-h/chopper#602]
**Is your feature request related to a problem? Please describe.**

I have multiple endpoints and must create multiple clients to interface with them, or define the baseUrl on the method level.

**Describe the solution you'd like**
_A clear and concise description of what you want to happen._

I want to configure one chopper client, where each service can hit a different endpoint.

**Describe alternatives you've considered**
The aforementioned workarounds

**Additional context**

**Comment by techouse** [comment]
>  I have multiple endpoints and must create multiple clients to interface with them, or define the baseUrl on the method level.

What **exactly** do you mean by "multiple endpoints"?

> I want to configure one chopper client, where each service can hit a different endpoint.

Isn't that already possible? I'm pretty sure I'm doing that in all my apps 🤷 The only time I'd create a different Chopper client is when I'd have to call a different API provider, i.e. something from `hxxps://api-one.com` and something `hxxps://api-two.com` where the 2 could have vastly different auth mechanism etc.

**Comment by techouse** [comment]
> This is what I am attempting to do with one Chopper client.

I don't understand why you would want a single API client to handle 2 different APIs? If this is what you want to achieve you'll probably have to do some additional abstraction.

>  Removing the base URL results in the app one being used from what I can tell.

Can you be a bit more specific, i.e. removing the `baseUrl` in the client or service level?

**Comment by techouse** [comment]
Yeah, this won't work and implementing such a feature would turn Chopper on its head.

**Comment by Guldem** [comment]
Maybe I don't fully understand it but if you want to define different baseUrl in the services then is possible. Not sure if it the nicest way. See [request docs](https://github.com/lejard-h/chopper/blob/develop/requests.md#path-resolution).

```dart
final chopper = ChopperClient(
    client: client,
    baseUrl: Uri.parse('http://localhost:8000'),
    // bind your object factories here
    converter: converter,
    errorConverter: converter,
    services: [
      UsersService.create(),
    ],
    /* Interceptors */
    interceptors: [AuthInterceptor()],
);

@ChopperApi()
abstract class UsersService extends ChopperService {
  static UsersService create([ChopperClient? client]) => _$UsersService(client);

  @Get(path: 'https://my-different-url/v1/api/users/{id}')
  Future<Response<User?>> getUser(@Path('id') String id);

  @Get(path: 'https://my-different-url/v2/api/users/{id}/articles')
  Future<Response<List<Article>>> getUserArticles(@Path('id') String id);

  @Delete(path: 'https://my-different-url/admin-v1/api/users')
  Future<Response> adminDeleteUser(@Query('id') String id);
}
```

While the clients baseUrl is localhost the services request all have `https://my-different-url/` specified in the request and will be used.

**Comment by techouse** [comment]
That's correct, because https://github.com/lejard-h/chopper/blob/fc6086382f6ed680e29255d7042102398d2705ca/chopper/lib/src/request.dart#L95
